File: MyJournal/app/src/main/java/com/smart/journal/module/write/bean/ToolBean.java
package com.smart.journal.module.write.bean;

import java.io.Serializable;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import androidx.annotation.StringDef;

/**
 * @author guandongchen
 * @date 2018/1/18
 */

public class ToolBean implements Serializable{

    @StringDef({ToolBeanType.TOOL_IMAGE,ToolBeanType.TOOL_LOCATION})
    @Retention(RetentionPolicy.SOURCE)
    public @interface ToolBeanType{
        /**
         * 图片
         */
        String TOOL_IMAGE = "Image";
        String TOOL_LOCATION = "Location";
        String TOOL_MORE = "More";
    }

    private int icon;
    private boolean selected;
    @ToolBeanType
    private String itemType;

    public ToolBean(int icon, boolean selected) {
        this.icon = icon;
        this.selected = selected;
    }

    public ToolBean(int icon,@ToolBeanType String itemType) {
        this.icon = icon;
        this.itemType = itemType;
    }

    public ToolBean(int icon) {
        this.icon = icon;
    }

    public ToolBean() {
    }

    public int getIcon() {
        return icon;
    }

    public void setIcon(int icon) {
        this.icon = icon;
    }

    public boolean isSelected() {
        return selected;
    }

    public void setSelected(boolean selected) {
        this.selected = selected;
    }

    public String getItemType() {
        return itemType;
    }

    public void setItemType(String itemType) {
        this.itemType = itemType;
    }
}

File: MyJournal/app/src/main/java/com/smart/journal/module/write/bean/ToolBean.kt
package com.smart.journal.module.write.bean

import androidx.annotation.StringDef
import com.smart.journal.module.write.bean.ToolBean.ToolBeanType.Companion.TOOL_IMAGE
import com.smart.journal.module.write.bean.ToolBean.ToolBeanType.Companion.TOOL_LOCATION
import java.io.Serializable


/**
 * @author guandongchen
 * @date 2018/1/18
 */
class ToolBean : Serializable {
    @Retention(AnnotationRetention.SOURCE)
    @StringDef(TOOL_IMAGE, TOOL_LOCATION)
    annotation class ToolBeanType {
        companion object {
            /**
             * 图片
             */
            const val TOOL_IMAGE = "Image"
            const val TOOL_LOCATION = "Location"
            const val TOOL_MORE = "More"
        }
    }

    var icon = 0
    var isSelected = false

    @ToolBeanType
    var itemType: String? = null

    constructor(icon: Int, selected: Boolean) {
        this.icon = icon
        isSelected = selected
    }

    constructor(icon: Int, @ToolBeanType itemType: String?) {
        this.icon = icon
        this.itemType = itemType
    }

    constructor(icon: Int) {
        this.icon = icon
    }

    constructor() {}

}